design/kanister-kopia-integration/kanister-kopia-integration.md

```diff
@@ -0,0 +1,91 @@
+Although they are useful and simple to use, these Functions can be significantly improved to provide better reliability, security, and performance.
+
+The improvements would include:
+1. Encryption of data during transfers and at rest
+2. Efficient content-based data deduplication
+3. Configurable data compression
+4. Reduced memory consumption
+5. Increased variety of backend storage target for backups
+
+These improvements can be achieved by using `Kopia` as the primary data movement tool in these Kanister Functions.
+
+Kanister also provides a command line utility `kando` that can be used to move data to and from object stores.
+This tool internally executes `Kopia` commands to move the backup data.
+The v2 version of the example Kanister Blueprints supports this. However, there are a few caveats to using these Blueprints.
+1. `kando` uses `Kopia` only when a Kanister Profile of type `Kopia` is provided
+2. A Kanister Profile of type `Kopia` requires a [Kopia Repository Server](https://kopia.io/docs/repository-server/) running in the same namespace as the Kanister controller
+3. A Repository Server requires a [Kopia Repository](https://kopia.io/docs/repositories/) to be initialized on a backend storage target
+
+Kanister currently lacks documentation and automation to use these features.
+
+## Introducing Kopia
+
+Kopia is a powerful, cross-platform tool for managing encrypted backups in the cloud.
+It provides fast and secure backups, using compression, data deduplication, and client-side end-to-end encryption.
+It supports a variety of backup storage targets, including object stores, which allows users to choose the storage provider that better addresses their needs. In Kopia, these storage locations are called repositories.
+It is a lock-free system that allows concurrent multi-client operations including garbage collection.
+
+To explore other features of Kopia, see its [documentation](https://kopia.io/docs/features/).
+
+## Scope
+
+1. Automate the initialization of a Kopia Repository for an application backed up by Kanister.
+2. Design and automate the lifecycle of the required Kopia Repository Server.
+3. Add new versions of Kanister Data Functions like BackupData, RestoreData, etc. with Kopia as the primary data mover tool.
+
+## User Experience
+
+- All the new features mentioned in this document will be opt-in only. Existing users will not see any changes in the Kanister controller's behavior.
+- Users will be able to continue using their current Blueprints, switch to the v2 version of the example Blueprints, or use Blueprints with the new version of the Kanister Data Functions.
+- Users opting to use the v2 Blueprints and Blueprints with Kopia-based Kanister Data Functions will be required to follow instructions to set up the required Kopia Repository Server before executing the actions.
```

Comment:
WDYT of adding a start-up option like `--storage=kopia`, `--feature-gate=storage=kopia` etc., which will enable the new CRD and auto-provision the default single kopia server?

Reply:
Do you mean the start-up of the controller? Sure, we can have a feature gate to enable the new CRD.
But note that the Blueprints are different for Kopia-based and stow/restic-based functions. If folks want to turn off the feature and use kopia blueprints, it won't be possible.